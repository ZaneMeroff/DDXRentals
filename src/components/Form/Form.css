#user-login-form {
  background-color: rgba(255, 255, 255, 0.8);
  border-radius: 35px;
  height: 450px;
  width: 40%;
  display: flex;
  flex-direction: column;
  justify-content: center;
  margin-left: 45%;
  margin-top: 100px;
}

.user-inputs {
  display: flex;
  flex-direction: row;
  justify-content: space-around;
  margin: 20px;
}

.form-input {
  height: 45px;
  width: 100%;
  margin: 10px;
  font-size: 1.5em;
  color: #DB49CB;
  padding: 5px;
}

#logo-img {
  position: absolute;
  height: auto;
  width: 400px;
  top: -40px;
  left: 40%;
}

#title-img {
  height: auto;
  width: 90%;
  margin-left: 5%;
  margin-top: 150px;

}

.radio-btn {
  margin-right: -30px;
  margin-top: 10px;
}

.btn-label {
  color: #DB49CB;
  font-size: 1.5em;
}

#form-btn {
  width: 200px;
  height: 50px;
  background-color: #FFBF03;
  font-size: 1.5em;
  color: #fff;
  border-radius: 15px;
}
