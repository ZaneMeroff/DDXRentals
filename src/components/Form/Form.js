import React, { Component } from 'react';
import logo from '../../data/DDR-logo.png';
import title from '../../data/DDR-title.png';
import './Form.css';

class Form extends Component {
  constructor() {
    super();
    this.state = {
      name: '',
      email: '',
      value: '',
      check: false
    }
  }
  handleChange = event => {
    this.setState({ [event.target.name]: event.target.value })
  }
  handleChoice = event => {
    // value here is 'on' when clicked
    // need to figure out how to turn all to off and then setState so user can only choose one
    // console.log(prevState);
    console.log(event);
    this.setState({ value: event.target.value });
  }
  render() {
    return (
      <form id='user-login-form'>
        <img src={logo} alt='DDR Xtreme logo' id='logo-img'/>
        <img src={title} alt='DDR Xtreme title' id='title-img'/>
        <div className='user-inputs'>
          <input placeholder='Name' name='name' className='form-input' value={this.state.name} onChange={this.handleChange} autoComplete='off'/>
          <input placeholder='Email' name='email' className='form-input' value={this.state.email} onChange={this.handleChange} autoComplete='off'/>
        </div>
        <div className='user-inputs dropdown'>
          <h2 className=''>Choose Your Adventure Type</h2>
          <select onChange={this.handleChoice} className='btn-label' value={this.state.value}>
            <option value='business'>Business</option>
            <option value='recreational'>Recreational</option>
            <option value='extreme'>Xtreme</option>
          </select>
        </div>
        <div className='user-inputs'>
          <button type='button' id='form-btn'>Send It</button>
        </div>
      </form>
    )
  }
}
export default Form;
