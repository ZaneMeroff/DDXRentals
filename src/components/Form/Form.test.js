import React from 'react';
import ReactDOM from 'react-dom';
import Form from './Form';
import { shallow } from 'enzyme';

describe('Form', () => {
  let mockUserInfo, wrapper;

  beforeEach(() => {
    mockUserInfo = { email: 'justhangin@aol.com', name: 'Billy', value: 'Business' };
  });

  it('should be an instance of Form component', () => {
    wrapper = shallow(<Form
        saveUserData={mockUserInfo}
      />);
      expect(wrapper).toMatchSnapshot();
  });

  it('should update its state when handleChange is invoked', () => {
    wrapper.instance().handleChange = jest.fn();
    wrapper.instance().handleChange();
    expect(wrapper.instance().handleChange).toHaveBeenCalled();
  });

  it('should update its state when handleChoice is invoked', () => {
    wrapper.instance().handleChoice = jest.fn();
    wrapper.instance().handleChoice();
    expect(wrapper.instance().handleChoice).toHaveBeenCalled();
  });

});
