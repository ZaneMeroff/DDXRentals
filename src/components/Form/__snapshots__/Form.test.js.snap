// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form should be an instance of Form component 1`] = `
<main
  className="app-home"
>
  <form
    id="user-login-form"
    onSubmit={[Function]}
  >
    <img
      alt="DDR Xtreme logo"
      id="logo-img"
      src="DDR-logo.png"
    />
    <img
      alt="DDR Xtreme title"
      id="title-img"
      src="DDR-title.png"
    />
    <div
      className="user-inputs"
    >
      <input
        autoComplete="off"
        className="form-input"
        name="name"
        onChange={[Function]}
        placeholder="Name"
        required={true}
        value=""
      />
      <input
        autoComplete="off"
        className="form-input"
        name="email"
        onChange={[Function]}
        placeholder="Email"
        required={true}
        type="email"
        value=""
      />
    </div>
    <div
      className="user-inputs dropdown"
    >
      <h2
        className=""
      >
        Choose Your Adventure Type
      </h2>
      <select
        className="btn-label"
        onChange={[Function]}
        value="business"
      >
        <option
          value="Business"
        >
          Business
        </option>
        <option
          value="Recreational"
        >
          Recreational
        </option>
        <option
          value="EXTREME"
        >
          Xtreme
        </option>
      </select>
    </div>
    <div
      className="user-inputs"
    >
      <input
        id="form-btn"
        type="submit"
        value="Send it!"
      />
    </div>
  </form>
</main>
`;
